VueJS/index.js

```diff
@@ -1,13 +1,84 @@
-import List from './components/List.js'
-import ListItem from './components/ListItem.js'
+Vue.component('todo-list', {
+	template: `<div>
+		<ul>
+			<todo-item
+				v-for="(todo, index) in todos"
+				v-bind:todo="todo"
+				v-on:save="$emit('save', {index: index, message: $event})"
+				v-on:remove="$emit('remove', index)"
+			/>\
+		</ul>
+		<button v-on:click="$emit('add')">Add new todo</button>
+		</div>`,
+	props: ['todos']
+})
+
+Vue.component('todo-item', {
+  template: `<li>
+	<template v-if="editing">
+		{{ todo.id }} {{ todo.message }}
+		<button v-on:click="editing=false">Edit</button>
+	</template>
+	<template v-else>
+		<input placeholder="Type todo here..." v-model="input"/>
+		<button v-on:click="save">Save</button>\
+		<button v-on:click="editing=true">Cancel</button>
+	</template>
+	<button v-on:click="remove">Remove</button>
+    </li>`,
+  props: ['todo'],
+  data: function() {
+	return {
+		editing: !(this.todo.message === ""),
+		input: this.todo.message,
+		identifier: this.todo.id
+	}
+  },
+  methods: {
+	save: function() {
+		this.$emit('save', this.input);
+		this.editing = true;
+	},
+	remove: function() {
+		this.$emit('remove', this.identifier)
+	}
+  }
+})
+
+new Vue({ 
+    el: '#hello',
+    data: {
+        message: 'Hello Vue.js!'
+    }
+});
 
 new Vue({
   el: '#app',
   data: {
+	cId: -1,
```

Comment:
This is probably the 6th time that I write:

> State is the root of all evil

At the very same location. Have a look into the other PRs to get an idea how to avoid this unnecessary counter, that needs to be manually updated. E.g. [here](https://github.com/m-oehme/Systems-Development-and-Frameworks/pull/2#discussion_r339841486)